ouroboros-consensus-cardano/test/Test/ThreadNet/Cardano.hs

```diff
@@ -237,15 +237,25 @@ mkProtocolCardano pbftParams coreNodeId genesisByron generatedSecretsByron
           generatedSecretsByron
           coreNodeId
 
+    -- the protocol version that each Byron node is endorsing with each block
+    -- it forges (ie which the node is ready to run)
+    --
+    -- Byron on mainnet is currently 0.2.0; Byron as configured in this test is
```

Comment:
Ah, I had only read the top of that thread.

So: the "Byron to Shelley" hard fork will switch between 1.0.0 and 2.0. (OBFT is 1.0.0,the latest "mini-era" within the Byron era.)

Reply:
OK. We sorted this out on Slack.

Mainnet: Classic 0.2.0; OBFT 1.0.0: Shelley 2.0

In this test: OBFT 0.0.0: Shelley 1.0

The ledgers both default to initially being 0. In this test, we're skipping the Classic part of the Byron era, so OBFT is major version 0. It would be inconvenient/excessive to be able to override the initial Byron ledger's value here.

And we want only one protocol update to suffice to switch to Shelley, but the Byron ledger requires an update can only increase the major version by 1. So in this test Shelley is therefore major version 1.

In order to have production and this test use different major versions for different eras etc, I had to add some parameters to Shelley initialization functions. It was tenable. See the new diff.